### PR TITLE
Correct the docstring of `EnergyDispersion2D`

### LINKED ISSUE #5530: Docstring of `EnergyDispersion2D` lists wrong parameters
**Gammapy version**
1.2, dev

**Bug description**
The docstring of `EnergyDispersion2D` lists the following parameters for its constructor:
```
    Parameters
    ----------
    energy_axis_true : `MapAxis`
        True energy axis.
    migra_axis : `MapAxis`
        Energy migration axis.
    offset_axis : `MapAxis`
        Field of view offset axis.
```
However – presumably because of a change to the `IRF` class? – the constructor does not accept a parameter `energy_axis_true`:
```
EnergyDispersion2D(energy_axis_true=0)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 EnergyDispersion2D(energy_axis_true=0)

TypeError: IRF.__init__() got an unexpected keyword argument 'energy_axis_true'
```

So I assume that the docstring of this class should be updated, and possible of other classes in `irf` too.

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR resolves #5530
It also adjusts the other docstrings where `axes` is utilised

### CODE REVIEW

File: gammapy/irf/edisp/kernel.py [hunk 1]
```diff
@@ -233,7 +233,10 @@ def from_hdulist(cls, hdulist, hdu1="MATRIX", hdu2="EBOUNDS"):
                     chan_max = l.field("F_CHAN")[k] + l.field("N_CHAN")[k]
 
                     pdf_matrix[i, chan_min:chan_max] = l.field("MATRIX")[
-                        m_start : m_start + l.field("N_CHAN")[k]  # noqa: E203
+                        m_start : m_start
+                        + l.field("N_CHAN")[
+                            k
+                        ]  # noqa: E203
```
Comment (morcuended) [inline gammapy/irf/edisp/kernel.py:239]: Was this changed by the precommit hook despite the noqa annotation? I think it is better as it was.

File: gammapy/irf/psf/parametric.py [hunk 2]
```diff
@@ -300,8 +300,11 @@ class EnergyDependentMultiGaussPSF(ParametricPSF):
 
     Parameters
     ----------
-    axes : list of `MapAxis`
-        Required axes are ["energy_true", "offset"].
+    axes : list of `~gammapy.maps.MapAxis` or `~gammapy.maps.MapAxes`
+        Required axes (in the given order) are:
+            * energy_true (true energy axis.)
+            * migra_axis (energy migration axis.)
+            * offset_axis (field of view offset axis.)
```
Comment (morcuended) [inline gammapy/irf/psf/parametric.py:307]: remove periods inside parenthesis (no periods in the rest of docstrings)

**Review by morcuended — COMMENTED** [review]
Looks good, some minor comments

File: gammapy/irf/edisp/kernel.py (hunk 1 above)
Comment (Astro-Kirsty) [reply gammapy/irf/edisp/kernel.py:239]: Yes it must have been, I did not adjust that

File: gammapy/irf/psf/parametric.py (hunk 2 above)
Comment (morcuended) [inline gammapy/irf/psf/parametric.py:307]:
```suggestion
            * energy_true (true energy axis)
            * migra_axis (energy migration axis)
            * offset_axis (field of view offset axis)
```

**Review by morcuended — APPROVED** [review]
Thanks @Astro-Kirsty. No further comments from my side. Regarding the automatic format change we can leave it as it was changed (and later decide how to deal with that rule in the ruff config)

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty

**Pull request merged**